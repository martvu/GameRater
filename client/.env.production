VITE_SERVER_URI=http://it2810-48.idi.ntnu.no:4000/graphql