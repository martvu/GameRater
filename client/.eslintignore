dist
.eslintrc.js
.prettierrc.js
src/gql/