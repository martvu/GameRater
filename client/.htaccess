<IfModule mod_rewrite.c>
  RewriteEngine On
  RewriteBase /project2/
  RewriteRule ^index\.html$ - [L]
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteRule . /project2/index.html [L]
</IfModule>
