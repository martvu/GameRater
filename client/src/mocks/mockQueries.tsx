import {
  ADD_FAVORITES,
  CREATE_REVIEW,
  REMOVE_FAVORITES,
  SIGN_IN_OR_CREATE_USER,
} from '@/lib/mutations';
import {
  GET_GAMES,
  GET_GAME,
  GET_FILTERS,
  GET_GAME_PLATFORMS,
  GET_SEARCH_SUGGESTIONS,
} from '@/lib/queries';

const getGamesMock = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: undefined,
      userId: '',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            user_rating: 4.5,
          },
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getGamesMockedKeyword = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: 'mockedKeyword',
      userId: '',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            user_rating: 4.5,
          },
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getGamesSignedInMock = {
  request: {
    query: GET_GAMES,
    variables: {
      limit: 24,
      offset: 0,
      sortBy: { field: 'first_release_date', order: 'desc' },
      platforms: [],
      genres: [],
      query: 'mockedKeyword',
      userId: '1',
      showFavorites: false,
      showReviewedGames: false,
    },
  },
  result: {
    data: {
      search: {
        count: 1,
        games: [
          {
            _id: '1',
            aggregated_rating: 85,
            first_release_date: 1620000000,
            summary: 'Example game summary',
            cover_image_id: 'cover1',
            name: 'Example Game',
            platforms: [1, 2],
            genres: [3, 4],
          },
          // Add more game objects as needed
        ],
        filters: {
          genres: [1, 2], // Example genre IDs
          platforms: [3, 4], // Example platform IDs
        },
      },
    },
  },
};

const getFiltersMock = {
  request: {
    query: GET_FILTERS,
    variables: {},
  },
  result: {
    data: {
      getGenres: [
        { id: 1, name: 'Genre 1' },
        { id: 2, name: 'Genre 2' },
      ],
      getPlatforms: [
        { id: 1, name: 'Platform 1' },
        { id: 3, name: 'Platform 3' },
        { id: 4, name: 'PC (Microsoft Windows)' },
        { id: 2, name: 'Nintendo Switch' },
        { id: 5, name: 'PlayStation 5' },
        { id: 6, name: 'Xbox Series X|S' },
      ],
    },
  },
};

const getGamePlatformsMock = {
  request: {
    query: GET_GAME_PLATFORMS,
    variables: { id: '1' },
  },
  result: {
    data: {
      getGame: {
        name: 'Example Game',
        platforms: [
          {
            id: 3,
            name: 'Platform 3',
          },
          {
            id: 4,
            name: 'PC (Microsoft Windows)',
          },
        ],
      },
    },
  },
};

const signInMock = {
  request: {
    query: SIGN_IN_OR_CREATE_USER,
    variables: {
      userInput: {
        username: 'testuser',
      },
    },
  },
  result: {
    data: {
      signInOrCreateUser: {
        _id: '1',
        username: 'testuser',
        favorites: [],
        reviews: [],
      },
    },
  },
};

const getGameMock = {
  request: {
    query: GET_GAME,
    variables: {
      id: '1',
      limit: 5,
      offset: 0,
      username: '',
    },
  },
  result: {
    data: {
      getGame: {
        _id: '1',
        name: 'Example Game',
        summary: 'This is an example game summary.',
        imageId: 'image1',
        first_release_date: 1620000000,
        aggregatedRating: 85,
        user_rating: 4.5,
        platforms: [{ name: 'PC' }],
        genres: [{ name: 'Adventure' }],
        reviews: {
          count: 2,
          reviews: [
            {
              _id: 'review1',
              user: 'user1',
              title: 'Great Game',
              content: 'This is a great game because...',
              rating: 5,
              platform: 'PC',
              gameID: '1',
            },
            {
              _id: 'review2',
              user: 'user2',
              title: 'Good Game',
              content: 'This is a good game because...',
              rating: 4,
              platform: 'PC',
              gameID: '1',
            },
          ],
          userHasReviewed: false,
        },
      },
    },
  },
};

const getGameMock2 = {
  request: {
    query: GET_GAME,
    variables: {
      id: '1',
      limit: 5,
      offset: 0,
      username: 'testUser',
    },
  },
  result: {
    data: {
      getGame: {
        _id: '1',
        name: 'Example Game',
        summary: 'This is an example game summary.',
        imageId: 'image1',
        first_release_date: 1620000000,
        aggregatedRating: 85,
        user_rating: 4.5,
        platforms: [{ name: 'PC' }],
        genres: [{ name: 'Adventure' }],
        reviews: {
          count: 2,
          reviews: [
            {
              _id: 'review1',
              user: 'user1',
              title: 'Great Game',
              content: 'This is a great game because...',
              rating: 5,
              platform: 'PC',
              gameID: '1',
            },
            {
              _id: 'review2',
              user: 'user2',
              title: 'Good Game',
              content: 'This is a good game because...',
              rating: 4,
              platform: 'PC',
              gameID: '1',
            },
          ],
          userHasReviewed: false,
        },
      },
    },
  },
};
const addFavoritesMock = {
  request: {
    query: ADD_FAVORITES,
    variables: { username: 'testUser', gameID: '1' },
  },
  result: {
    data: {
      addFavorites: {
        _id: '1',
        username: 'testUser',
        favorites: [{ _id: '1' }],
      },
    },
  },
};

const removeFavoritesMock = {
  request: {
    query: REMOVE_FAVORITES,
    variables: { username: 'testUser', gameID: '1' },
  },
  result: {
    data: {
      removeFavorites: {
        _id: '1',
        username: 'testUser',
        favorites: [],
      },
    },
  },
};

const createReviewMock = {
  request: {
    query: CREATE_REVIEW,
    variables: {
      reviewInput: {
        title: 'Great Game',
        content: 'This is a great game because...',
        rating: 5,
        platform: 'PC (Microsoft Windows)',
        user: 'testUser',
        gameID: '1',
      },
    },
  },
  result: {
    data: {
      createReview: {
        _id: 'review1',
        user: 'testUser',
        title: 'Great Game',
        content: 'This is a great game because...',
        rating: 5,
        platform: 'PC (Microsoft Windows)',
        gameID: '1',
      },
    },
  },
};

const searchbarSuggestionsMock = {
  request: {
    query: GET_SEARCH_SUGGESTIONS,
    variables: {
      query: 'example',
    },
  },
  result: {
    data: {
      getSearchSuggestions: [
        {
          _id: '1',
          name: 'Example Game',
          cover_image_id: 'cover1',
        },
        {
          _id: '2',
          name: 'Example Game 2',
          cover_image_id: 'cover2',
        },
      ],
    },
  },
};

const allMocks = [
  createReviewMock,
  getGamePlatformsMock,
  getGamesMockedKeyword,
  addFavoritesMock,
  removeFavoritesMock,
  getGamesMock,
  getFiltersMock,
  signInMock,
  getGamesSignedInMock,
  getGameMock,
  getGameMock2,
  searchbarSuggestionsMock,
  // Add more mocks as needed for other queries
];

export {
  allMocks,
  getGamesMock,
  getGamesSignedInMock,
  getFiltersMock,
  signInMock,
};
