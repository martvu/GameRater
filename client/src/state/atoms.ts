import { atom } from 'recoil';
import { GameSortInput, User } from '../gql/graphql';

const user = localStorage.getItem('user');
const defaultUser = {
  _id: '',
  username: '',
  favorites: [],
  reviews: [],
};

export const userState = atom<User>({
  key: 'userState',
  default: user ? JSON.parse(user) : defaultUser,
});

export const selectedPlatformsState = atom({
  key: 'selectedPlatformsState',
  default: [] as number[],
});

export const selectedGenresState = atom({
  key: 'selectedGenresState',
  default: [] as number[],
});

const currentPage = localStorage.getItem('currentPage');
export const pageState = atom<number>({
  key: 'pageState',
  default: currentPage ? parseInt(currentPage, 10) : 1,
});

const sortBy = localStorage.getItem('selectedSortState');

export const defaultSortBy = {
  field: 'first_release_date',
  order: 'desc',
};

export const sortByState = atom<GameSortInput>({
  key: 'sortByState',
  default: sortBy ? JSON.parse(sortBy) : defaultSortBy,
});

