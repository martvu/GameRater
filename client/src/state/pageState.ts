import { atom } from 'recoil';

const currentPage = localStorage.getItem('currentPage');
console.log("state", currentPage);
const defaultPage = 1;

export const pageState = atom<number>({
  key: 'pageState',
  default: currentPage ? parseInt(currentPage, 10) : defaultPage,
});
