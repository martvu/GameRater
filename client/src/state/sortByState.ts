import { atom } from 'recoil';
import { GameSortInput } from '../gql/graphql';

const sortBy = localStorage.getItem('selectedSortState');

export const defaultSortBy = {
  field: 'first_release_date',
  order: 'desc',
};

export const sortByState = atom<GameSortInput>({
  key: 'sortByState',
  default: sortBy ? JSON.parse(sortBy) : defaultSortBy,
});
