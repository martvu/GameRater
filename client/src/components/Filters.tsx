import { useState } from 'react';
import { ChevronLeft, ListFilter } from 'lucide-react';
import FilterItems from './FilterItems';
import { Button } from './ui/button';
import { cn } from '@/lib/utils';
import { GET_FILTERS } from '@/lib/queries';
import { useQuery } from '@apollo/client';
import { Genre, Platform } from '@/gql/graphql';
import { useRecoilValue } from 'recoil';
import {
  genresListState,
  platformsListState,
  selectedGenresState,
  selectedPlatformsState,
} from '@/state/atoms';
import { ScrollArea } from './ui/scroll-area';

/**
 * Filters component
 * Displayed on large screens (non-mobile)
 */
export default function Filters() {
  const [isCollapsed, setIsCollapsed] = useState(false);
  const genreList = useRecoilValue(genresListState);
  const platformList = useRecoilValue(platformsListState);
  const selectedGenres = useRecoilValue(selectedGenresState);
  const selectedPlatforms = useRecoilValue(selectedPlatformsState);

  const toggleCollapse = () => {
    setIsCollapsed(!isCollapsed);
  };
  const { data } = useQuery(GET_FILTERS);

  const preferredOrder = [
    'Nintendo Switch',
    'PC (Microsoft Windows)',
    'PlayStation 5',
    'Xbox Series X|S',
  ];

  // Custom sorting function to sort platforms by preferred order
  const sortByPreference = (a: Platform, b: Platform) => {
    const aIndex = preferredOrder.indexOf(a.name as string);
    const bIndex = preferredOrder.indexOf(b.name as string);

    if (aIndex !== -1 && bIndex !== -1) {
      return aIndex - bIndex; // Both are preferred, sort by preferred order
    } else if (aIndex !== -1) {
      return -1; // Only a is preferred
    } else if (bIndex !== -1) {
      return 1; // Only b is preferred
    }
    return 0; // Neither is preferred, maintain original order
  };

  // Sort the platforms
  // Create a copy of the platforms array before sorting
  const platformsCopy: Platform[] = [...(data?.getPlatforms ?? [])];
  const genresCopy: Genre[] = [...(data?.getGenres ?? [])];
  // Now sort the copied array
  const sortedPlatforms: Platform[] = platformsCopy.sort(sortByPreference);

  // Filter only platforms in platformList (when search limits platforms) or all platforms if platformList is empty
  const filteredPlatforms = sortedPlatforms.filter(platform => {
    if (platformList.length === 0) return true;
    return (
      platformList.includes(platform.id as number) ||
      selectedPlatforms.includes(platform.id as number)
    );
  });

  // Filter only genres in genreList or all genres if genreList is empty
  const filteredGenres = genresCopy.filter(genre => {
    if (genreList.length === 0) return true;
    return (
      genreList.includes(genre.id as number) ||
      selectedGenres.includes(genre.id as number)
    );
  });

  // Apply transition to the filter menu
  const filterContainerClasses = cn(
    'flex h-full flex-col items-start overflow-hidden text-left transition-all duration-300 ease-in-out',
    isCollapsed ? 'w-0' : 'w-[240px]' // Toggle between full width and collapsed state
  );

  return (
    <>
      <section className={filterContainerClasses}>
        <ScrollArea className="h-full pb-4 pr-2">
          <section className="pb-4 pl-5 md:pb-32">
            <h1
              className={`mt-4 text-left text-xl font-bold tracking-wider text-foreground ${
                isCollapsed ? 'hidden' : ''
              }`}
            >
              Filter By
            </h1>
            {!isCollapsed && data && (
              <>
                <FilterItems
                  filters={filteredPlatforms}
                  filterType="platforms"
                />
                <FilterItems filters={filteredGenres} filterType="genres" />
              </>
            )}
          </section>
        </ScrollArea>
      </section>
      <Button
        className="fixed bottom-8 left-8 z-50 m-1 hidden md:flex"
        variant="secondary"
        onClick={toggleCollapse}
        aria-label={isCollapsed ? 'show filters' : 'hide filters'}
      >
        {isCollapsed ? <ListFilter /> : <ChevronLeft size={18} />}
        {!isCollapsed && <span className="ml-2">Hide filters</span>}
      </Button>
    </>
  );
}
