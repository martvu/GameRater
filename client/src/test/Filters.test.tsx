import { describe, it, expect } from 'vitest';
import { render, screen, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { RecoilRoot } from 'recoil';
import Filters from '@/components/Filters'; // Adjust the path to your Filters component
import { allMocks } from '@/mocks/mockQueries';
import userEvent from '@testing-library/user-event';
import { platformsListState, genresListState } from '@/state/atoms';

interface RenderFiltersOptions {
  mockPlatformList?: number[];
  mockGenreList?: number[];
}

function renderFilters({
  mockPlatformList = [],
  mockGenreList = [],
}: RenderFiltersOptions = {}) {
  return render(
    <MockedProvider mocks={allMocks} addTypename={false}>
      <RecoilRoot
        initializeState={({ set }) => {
          set(platformsListState, mockPlatformList);
          set(genresListState, mockGenreList);
        }}
      >
        <Filters />
      </RecoilRoot>
    </MockedProvider>
  );
}

describe('Filters Component', () => {
  it('matches the snapshot', async () => {
    const { asFragment } = renderFilters();

    await waitFor(() => {
      expect(screen.getByText('Filter By')).toBeInTheDocument();
      expect(screen.getByText('platforms')).toBeInTheDocument();
      expect(screen.getByText('genres')).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly and handles toggle collapse', async () => {
    renderFilters();
    expect(screen.getByText('Filter By')).toBeInTheDocument();

    await waitFor(() => {
      expect(screen.getByText('platforms')).toBeInTheDocument();
      expect(screen.getByText('genres')).toBeInTheDocument();
      const checkboxes = screen.getAllByRole('checkbox');
      expect(checkboxes).toHaveLength(8);
    });

    // Test toggle collapse
    await userEvent.click(screen.getByLabelText('toggle filters'));

    await waitFor(() => {
      expect(screen.queryByText('platforms')).not.toBeInTheDocument();
      expect(screen.queryByText('genres')).not.toBeInTheDocument();
      const checkboxes = screen.queryAllByRole('checkbox');
      expect(checkboxes).toHaveLength(0);
    });
  });

  it('renders correctly when platformList and genreList is not empty', async () => {
    renderFilters({ mockPlatformList: [1, 2, 3], mockGenreList: [1] });
    await waitFor(() => {
      expect(screen.getByText('platforms')).toBeInTheDocument();
      const checkboxes = screen.getAllByRole('checkbox');
      expect(checkboxes).toHaveLength(4);
    });
  });

  it('checks the checkboxes correctly', async () => {
    renderFilters();
    await waitFor(() => {
      const checkboxes = screen.getAllByRole('checkbox');
      expect(checkboxes).toHaveLength(8);
    });
    const checkboxes = screen.getAllByRole('checkbox');
    await userEvent.click(checkboxes[0]);
    const genreCheckbox = screen.getByTestId('filter-item-Genre 1');
    await userEvent.click(genreCheckbox);
    await waitFor(() => {
      expect(checkboxes[0]).toBeChecked();
      expect(genreCheckbox).toBeChecked();
    });
  });
});
