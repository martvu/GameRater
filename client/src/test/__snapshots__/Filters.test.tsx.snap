// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filters Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex h-full flex-col items-start overflow-hidden text-left transition-all duration-300 ease-in-out w-[240px]"
  >
    <div
      class="relative overflow-hidden h-full pb-4 pr-2"
      dir="ltr"
      style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
    >
      <style>
        [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
      </style>
      <div
        class="h-full w-full rounded-[inherit]"
        data-radix-scroll-area-viewport=""
        style="overflow-x: hidden; overflow-y: scroll;"
      >
        <div
          style="min-width: 100%; display: table;"
        >
          <div
            class="pb-4 pl-5 md:pb-32"
          >
            <h1
              class="mt-4 text-left text-xl font-bold tracking-wider text-foreground "
            >
              Filter By
            </h1>
            <h2
              class="my-2 text-left font-semibold capitalize text-foreground"
            >
              platforms
            </h2>
            <div
              class="ml-1 flex flex-col"
            >
              <div
                class="my-1 flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="filter-item-Nintendo Switch"
                  class="peer h-5 w-5 shrink-0 rounded-sm border border-primary shadow-lg ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                  data-state="unchecked"
                  data-testid="filter-item-Nintendo Switch"
                  id="filter-item-Nintendo Switch"
                  role="checkbox"
                  type="button"
                  value="on"
                />
                <label
                  class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 line-clamp-1 text-left font-normal tracking-tight text-muted-foreground"
                  for="filter-item-Nintendo Switch"
                >
                  Nintendo Switch
                </label>
              </div>
              <div
                class="my-1 flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="filter-item-PC (Microsoft Windows)"
                  class="peer h-5 w-5 shrink-0 rounded-sm border border-primary shadow-lg ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                  data-state="unchecked"
                  data-testid="filter-item-PC (Microsoft Windows)"
                  id="filter-item-PC (Microsoft Windows)"
                  role="checkbox"
                  type="button"
                  value="on"
                />
                <label
                  class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 line-clamp-1 text-left font-normal tracking-tight text-muted-foreground"
                  for="filter-item-PC (Microsoft Windows)"
                >
                  PC (Microsoft Windows)
                </label>
              </div>
              <div
                class="my-1 flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="filter-item-PlayStation 5"
                  class="peer h-5 w-5 shrink-0 rounded-sm border border-primary shadow-lg ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                  data-state="unchecked"
                  data-testid="filter-item-PlayStation 5"
                  id="filter-item-PlayStation 5"
                  role="checkbox"
                  type="button"
                  value="on"
                />
                <label
                  class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 line-clamp-1 text-left font-normal tracking-tight text-muted-foreground"
                  for="filter-item-PlayStation 5"
                >
                  PlayStation 5
                </label>
              </div>
              <div
                class="my-1 flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="filter-item-Xbox Series X|S"
                  class="peer h-5 w-5 shrink-0 rounded-sm border border-primary shadow-lg ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                  data-state="unchecked"
                  data-testid="filter-item-Xbox Series X|S"
                  id="filter-item-Xbox Series X|S"
                  role="checkbox"
                  type="button"
                  value="on"
                />
                <label
                  class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 line-clamp-1 text-left font-normal tracking-tight text-muted-foreground"
                  for="filter-item-Xbox Series X|S"
                >
                  Xbox Series X|S
                </label>
              </div>
              <div
                class="my-1 flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="filter-item-Platform 1"
                  class="peer h-5 w-5 shrink-0 rounded-sm border border-primary shadow-lg ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                  data-state="unchecked"
                  data-testid="filter-item-Platform 1"
                  id="filter-item-Platform 1"
                  role="checkbox"
                  type="button"
                  value="on"
                />
                <label
                  class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 line-clamp-1 text-left font-normal tracking-tight text-muted-foreground"
                  for="filter-item-Platform 1"
                >
                  Platform 1
                </label>
              </div>
              <div
                class="my-1 flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="filter-item-Platform 3"
                  class="peer h-5 w-5 shrink-0 rounded-sm border border-primary shadow-lg ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                  data-state="unchecked"
                  data-testid="filter-item-Platform 3"
                  id="filter-item-Platform 3"
                  role="checkbox"
                  type="button"
                  value="on"
                />
                <label
                  class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 line-clamp-1 text-left font-normal tracking-tight text-muted-foreground"
                  for="filter-item-Platform 3"
                >
                  Platform 3
                </label>
              </div>
            </div>
            <div />
            <h2
              class="my-2 text-left font-semibold capitalize text-foreground"
            >
              genres
            </h2>
            <div
              class="ml-1 flex flex-col"
            >
              <div
                class="my-1 flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="filter-item-Genre 1"
                  class="peer h-5 w-5 shrink-0 rounded-sm border border-primary shadow-lg ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                  data-state="unchecked"
                  data-testid="filter-item-Genre 1"
                  id="filter-item-Genre 1"
                  role="checkbox"
                  type="button"
                  value="on"
                />
                <label
                  class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 line-clamp-1 text-left font-normal tracking-tight text-muted-foreground"
                  for="filter-item-Genre 1"
                >
                  Genre 1
                </label>
              </div>
              <div
                class="my-1 flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="filter-item-Genre 2"
                  class="peer h-5 w-5 shrink-0 rounded-sm border border-primary shadow-lg ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
                  data-state="unchecked"
                  data-testid="filter-item-Genre 2"
                  id="filter-item-Genre 2"
                  role="checkbox"
                  type="button"
                  value="on"
                />
                <label
                  class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 line-clamp-1 text-left font-normal tracking-tight text-muted-foreground"
                  for="filter-item-Genre 2"
                >
                  Genre 2
                </label>
              </div>
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    aria-label="toggle filters"
    class="items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2 fixed bottom-8 left-8 z-50 m-1 hidden md:flex"
  >
    <svg
      class="lucide lucide-chevron-left"
      fill="none"
      height="18"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m15 18-6-6 6-6"
      />
    </svg>
    <span
      class="ml-2"
    >
      Hide filters
    </span>
  </button>
</DocumentFragment>
`;
